### PR TITLE
[#138323307]framework declaration overview

### PR DESCRIPTION
**Pull request by risicle**
Story https://www.pivotaltracker.com/story/show/138323307

Will probably add more information here soon.

### CODE REVIEW

File: app/main/views/frameworks.py [hunk 1]
```diff
@@ -273,6 +274,80 @@ def framework_start_supplier_declaration(framework_slug):
...
+    )
+
+    return render_template(
+        "frameworks/declaration_overview.html",
+        framework=framework,
+        supplier_framework=sf,
+        sections_errors=sections_errors,
+        validates=not any(errors for section, errors in sections_errors.values()),
+        framework_dates=content_loader.get_message(framework_slug, "dates"),
+    )
```
Comment (benvand) [inline app/main/views/frameworks.py]: This view needs to return the `status_code` too

Comment (risicle) [inline app/main/views/frameworks.py]: Oh look at me the django person.

Doesn't it return 200 by default? Will add it for clarity anyway.

Comment (TheDoubleK) [inline app/main/views/frameworks.py]: I think 200 is default.

**Comment by risicle** [comment]
And here's more information. Should comply to all requirements in story.
20170217declarationoverviewcomplete
20170217declarationoverviewemptynoprefill
20170217declarationoverviewreadytomd

File: app/main/views/frameworks.py [hunk 2]
```diff
@@ -274,6 +275,80 @@ def framework_start_supplier_declaration(framework_slug):
...
+
+    sf = data_api_client.get_supplier_framework_info(current_user.supplier_id, framework_slug)["frameworkInterest"]
+    # ensure our declaration is at least (probably) a dict
+    sf["declaration"] = sf.get("declaration") or {}
+
+    content = content_loader.get_manifest(framework_slug, 'declaration').filter(sf["declaration"])
+
+    # we must perform an actual validation for each section rather than rely on .answer_required as the latter won't
+    # take into account declarations custom question dependencies
+    sections_errors = OrderedDict(
```
Comment (benvand) [inline app/main/views/frameworks.py]: This very difficult to read.
The same can be achieved with:
```
-    sections_errors = OrderedDict(
-        (
-            section.slug,
-            (
-                section,
-                section.editable and get_validator(
-                    framework,
-                    content,
-                    sf["declaration"],
-                ).get_error_messages_for_page(section),
-            ),
-        )
-        for section in content.summary(sf["declaration"])
-    )
+    # Build error dict like {section_slug: (section, errors?)}
+    validator = get_validator(framework, content, sf["declaration"])
+    sections_errors = OrderedDict()
+    for section in content.summary(sf["declaration"]):
+        sections_errors[section.slug] = (section, section.editable and validator.get_error_messages_for_page(section))
```

Comment (risicle) [inline app/main/views/frameworks.py]: I much prefer declarative over imperative. With my one, I can immediately discard the possibility that the block has any side effects (unless it's written *very* strangely). In imperative style it's very hard to tell between many subtly different patterns which may or may not have side effects.

Comment (risicle) [inline app/main/views/frameworks.py]: Though adding a comment describing the format would be helpful.

**Comment by TheDoubleK** [comment]
:+1: from me once Travis is happy

**Pull request merged**